test/test_testing.py

```diff
@@ -2751,6 +2751,19 @@ def _test_push_and_pull_data(self, original_data, encoding, stream_class):
             received_data = infile.read()
         self.assertEqual(original_data, received_data)
 
+    def test_push_and_pull_larger_file(self):
+        # Intent: to ensure things work appropriately with larger files.
+        # Larger files may be sent/received in multiple chunks; this should help for
+        # checking that such logic is correct.
+        data_size = 2**20
+        original_data = "".join(chr(i % 128) for i in range(data_size)).encode('ascii')
+
+        client = self.client
+        client.push(self.prefix + '/test', original_data, encoding=None)
+        with client.pull(self.prefix + '/test', encoding=None) as infile:
+            received_data = infile.read()
+        self.assertEqual(original_data, received_data)
```

Comment:
This certainly looks like we should be covering all the code paths, but it certainly doesn't make sense in *my* head why we have a file that may be open or closed at any given moment.